crypto/asn1/a_gentm.c

```diff
@@ -88,6 +104,9 @@ int asn1_generalizedtime_to_tm(struct tm *tm, const ASN1_GENERALIZEDTIME *d)
      * digits.
      */
     if (a[o] == '.') {
+        if (strict)
+            /* RFC 5280 forbids fractional seconds */
+            goto err;
         if (++o > l)
             goto err;
```

Comment:
I think you're right, the check will never be true. Currently it's a 'add-then-check' mode while iterating the input time string, the check is done by comparing the value in a[o] so it works only on null-terminated strings. For this check, probably just change `++o > l` to `++o == l` is fine enough, to avoid unnecessary subsequent checks. But I would like to make this logic more clear to change it to a 'non-null-terminated-depending' style, thus we don't need to be bothered by if the `t->data` has a '\0' or not any more. And I consider this as a more careful and more secure way of dealing with strings.

The same situations exist as well as in UTC parsing function.